### PR TITLE
virtio-serial:Keep chardev open to re-use

### PR DESCRIPTION
**Pull request by nanliu-r**
ID: 1808936

Signed-off-by: Nana Liu <nanliu@redhat.com>

### CODE REVIEW

**Comment by nanliu-r** [comment]
Test results:

 (1/2) Host_RHEL.m8.u2.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.virtio_port_hotplug.re-use_chardev: PASS (314.17 s)
 (2/2) Host_RHEL.m8.u2.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_port_hotplug.re-use_chardev: PASS (458.19 s)

**Comment by nanliu-r** [comment]
@yanan-fu Please help to check this, thanks a lot.

File: qemu/tests/cfg/virtio_port_hotplug.cfg [hunk 1]
```diff
@@ -44,6 +44,17 @@
             sleep_time = 0.5
             serials += " vs1"
             serial_type_vs1 = virtserialport
+        - re-use_chardev:
```
Comment (yanan-fu) [inline qemu/tests/cfg/virtio_port_hotplug.cfg:47]: 1. case name contain both '-' and '_', better use '_'.
2. test case `unplug_port` reuse chardev too, so i think it is unreasonable this variant with same level.
It is better to use sub variants for `unplug_port`. Something like: `unplug_port.default`, `unplug_port.with_file_transfer/with_stress..`.

File: qemu/tests/virtio_port_hotplug.py [hunk 2]
```diff
@@ -35,6 +35,7 @@ def run(test, params, env):
     timeout = int(params.get("login_timeout", 360))
     module = params.get("modprobe_module")
     check_module = params.get_boolean("check_module", "yes")
+    bg_test = params.get_boolean("bg_test", "yes")
```
Comment (PaulYuuu) [inline qemu/tests/virtio_port_hotplug.py:38]: typo: yes
```python
get_boolean(self, key, default=False)
```

Comment (yanan-fu) [reply qemu/tests/virtio_port_hotplug.py:38]: @PaulYuuu The function `get_boolean` accept `yes`,`on`,`true`

Comment (PaulYuuu) [reply qemu/tests/virtio_port_hotplug.py:38]: The second parameter requires a bool type.
`value = self.get(key, "yes" if default else "no")`
```python
    def get_boolean(self, key, default=False):
        """
        Check if a config option is set to a default affirmation or not.

        :param key: config option key
        :type key: str
        :param bool default: whether to assume "yes" or "no" if param
                             does not exist (defaults to False/"no").
        :return whether option is set to 'yes'
        :rtype: bool
        """
        value = self.get(key, "yes" if default else "no")
        if value in ("yes", "on", "true"):
            return True
        if value in ("no", "off", "false"):
            return False
        raise ValueError("Cannot get boolean parameter value for %s: %s", key, value)
```

Comment (yanan-fu) [reply qemu/tests/virtio_port_hotplug.py:38]: @PaulYuuu ah, yes, my careless , thanks for point it out.

File: qemu/tests/cfg/virtio_port_hotplug.cfg (hunk 1 above)
Comment (nanliu-r) [reply qemu/tests/cfg/virtio_port_hotplug.cfg]: update and thanks.

**Comment by nanliu-r** [comment]
@xiagao Hi, could help to check this since the variants struct has changed? thanks a lot.

File: qemu/tests/virtio_port_hotplug.py (hunk 2 above)
Comment (nanliu-r) [reply qemu/tests/virtio_port_hotplug.py]: thanks and update.

**Comment by nanliu-r** [comment]
test pass,
 (1/4) Host_RHEL.m8.u2.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.virtio_port_hotplug: PASS (113.28 s)
 (2/4) Host_RHEL.m8.u2.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.virtio_port_hotplug.with_bg_test: PASS (413.50 s)
 (3/4) Host_RHEL.m8.u2.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_port_hotplug: PASS (221.57 s)
 (4/4) Host_RHEL.m8.u2.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_port_hotplug.with_bg_test: PASS (453.91 s)

**Comment by yanan-fu** [comment]
Acked-by: Yanan Fu <yfu@redhat.com>

File: qemu/tests/cfg/virtio_port_hotplug.cfg [hunk 3]
```diff
@@ -37,8 +37,13 @@
             unplug_chardev_vs1 = no
             unplug_chardev_vs2 = no
             filesize = 100
-            Linux:
-                filesize = 2000
+            variants:
+                - @default:
+                    bg_test = no
```
Comment (xiagao) [inline qemu/tests/cfg/virtio_port_hotplug.cfg:42]: Pls keep them alignment with line37 in py file.
yes/no 
true/false

Comment (yanan-fu) [reply qemu/tests/cfg/virtio_port_hotplug.cfg:42]: @xiagao  They are not same meaning, you can check the function `get_boolean`.

Comment (xiagao) [reply qemu/tests/cfg/virtio_port_hotplug.cfg:42]: Yes, just now missed get_boolean function.

Comment (nanliu-r) [reply qemu/tests/cfg/virtio_port_hotplug.cfg:42]: > Pls keep them alignment with line37 in py file.
> yes/no
> true/false

please check the `get_boolean(self, key, default=False)`,  'yes/no' is the `key`, True/False is the default `boolean` and we need to set the `key` value.

**Comment by nanliu-r** [comment]
> LGTM.
> Ack.

Thanks, and please help to change the auto script name which is virtio_port_hotplug.unplug_port in your test loop.

**Comment by xiagao** [comment]
> > LGTM.
> > Ack.
> 
> Thanks, and please help to change the auto script name which is `virtio_port_hotplug.unplug_port` before in your test loop.

Sure,when it's merged.

**Comment by yanan-fu** [comment]
Thanks all!
@nanliu-r @xiagao Just reminder, please remember to update the case name in test loops/sheet/polarion case.

**Pull request merged**